### ISSUE TITLE
[kafka] How to expose brokers on K3s?

### ISSUE THREAD

**Issue by b-abel** [bitnami/charts#7171]
I am running the Bitnami Kafka Helm chart on the [k3s](https://k3s.io/) Kubernetes distribution on a publicly accessible VM. The Kafka cluster has only one broker and must be available from the internet.

The Helm chart is configured as:

~~~yaml
replicaCount: 1
externalAccess:
  enabled: true
  service:
    type: LoadBalancer
    port: 9092
    nodePorts:
      - 32092
  autoDiscovery:
    enabled: true
serviceAccount:
  create: true
rbac:
  create: true
service:
  port: 9094
~~~

Now, consumers cannot connect from the outside. The reason seems to be the listener configuration because the command `kafkacat -b <domain.name.com>:9092` reports an IP address that I cannot access from the outside. I suspect it is an internal IP of the ~[Traefik](https://doc.traefik.io/traefik)~ K3s load balancer that k3s enables by default. 
The [Kafka chart documentation](https://github.com/bitnami/charts/tree/master/bitnami/kafka#accessing-kafka-brokers-from-outside-the-clusters) shows how to set the domain only for the NodePort configuration, which I didn't use because I want to use Kafka's default port 9092 externally. 
I would be happy if someone could point me to an easy option to achieve this.

**Comment by pablogalegoc** [comment]
Hi @b-abel!

If I understand correctly, you’ve installed k3s and provisioned a cluster on a VM with a public IP. I’m not an expert on k3s but I’ve seen in their [LB documentation](https://rancher.com/docs/k3s/latest/en/networking/#how-the-service-lb-works) that `If the Service LB runs on a node that has an external IP, it uses the external IP`. Could it be that it is being run on a node without that external IP and k3s is creating another, which isn’t accesible from the outside?

**Comment by b-abel** [comment]
Hi @pablogalegoc!

Thank you very much for your hint. I will check this and come back with the result.

**Comment by b-abel** [comment]
For reference: I solved my issue with a different solution. I did not want to set the IP address of the VM at the Kubernetes node, nor at configuring it at the Helm chart (because the IP might change in the future without me instantly noticing it). Instead, I set the _domain name_ in `externalAccess.service.loadBalancerIPs[0]`, which interestingly works. So, my configuration that works as expected looks like this:

~~~yaml
externalAccess:
  enabled: true
  service:
    type: "LoadBalancer"
    port: 9092
    loadBalancerIPs:
      - "<domain.name>"
  autoDiscovery:
    enabled: false
serviceAccount:
  create: false
rbac:
  create: false
~~~
